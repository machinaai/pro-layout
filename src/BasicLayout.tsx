import './BasicLayout.less';

import React, { CSSProperties, useContext, useEffect, useState } from 'react';
import { BreadcrumbProps as AntdBreadcrumbProps } from 'antd/es/breadcrumb';
import { Layout } from 'antd';
import classNames from 'classnames';
import warning from 'warning';
import useMergeValue from 'use-merge-value';
import { stringify } from 'use-json-comparison';
import useAntdMediaQuery from 'use-media-antd-query';
import Omit from 'omit.js';
import Header, { HeaderViewProps } from './Header';
import {
  MenuDataItem,
  MessageDescriptor,
  Route,
  RouterTypes,
  WithFalse,
} from './typings';
import { getPageTitleInfo, GetPageTitleProps } from './getPageTitle';
import defaultSettings, { PureSettings } from './defaultSettings';
import getLocales, { LocaleType } from './locales';
import { BaseMenuProps } from './SiderMenu/BaseMenu';
import Footer from './Footer';
import RouteContext from './RouteContext';
import SiderMenu from './SiderMenu';
import { SiderMenuProps } from './SiderMenu/SiderMenu';
import { getBreadcrumbProps } from './utils/getBreadcrumbProps';
import getMenuData from './utils/getMenuData';
import { isBrowser, useDeepCompareEffect } from './utils/utils';
import PageLoading from './PageLoading';
import MenuCounter from './SiderMenu/Counter';
import WrapContent from './WrapContent';
import { useDocumentTitle } from './utils/hooks';
import compatibleLayout from './utils/compatibleLayout';

export type BasicLayoutProps = Partial<RouterTypes<Route>> &
  SiderMenuProps &
  HeaderViewProps &
  Partial<PureSettings> & {
    pure?: boolean;
    /**
     * logo url
     */
    logo?: React.ReactNode | WithFalse<() => React.ReactNode>;

    /**
     * Se activa cuando se cambia la p√°gina
     */
    onPageChange?: (location?: RouterTypes<Route>['location']) => void;

    loading?: boolean;

    locale?: LocaleType;

    onCollapse?: (collapsed: boolean) => void;

    footerRender?: WithFalse<
      (props: HeaderViewProps, defaultDom: React.ReactNode) => React.ReactNode
    >;

    breadcrumbRender?: (
      routers: AntdBreadcrumbProps['routes'],
    ) => AntdBreadcrumbProps['routes'];
    menuItemRender?: BaseMenuProps['menuItemRender'];
    pageTitleRender?: WithFalse<
      (
        props: GetPageTitleProps,
        defaultPageTitle?: string,
        info?: {
          // T√≠tulo de la p√°gina
          title: string;
          // local de t√≠tulo
          id: string;
          // T√≠tulo de la p√°gina sin valor predeterminado
          pageName: string;
        },
      ) => string
    >;
    menuDataRender?: (menuData: MenuDataItem[]) => MenuDataItem[];
    itemRender?: AntdBreadcrumbProps['itemRender'];

    formatMessage?: (message: MessageDescriptor) => string;
    /**
     * Ya sea que deshabilite el modo m√≥vil, algunos sistemas de administraci√≥n no necesitan el modo m√≥vil, esta propiedad se puede establecer en true
     */
    disableMobile?: boolean;
    contentStyle?: CSSProperties;
    isChildrenLayout?: boolean;

    className?: string;

    /**
     * Use ambos m√°rgenes de contenido
     */
    disableContentMargin?: boolean;
  };

const headerRender = (
  props: BasicLayoutProps & {
    hasSiderMenu: boolean;
  },
): React.ReactNode => {
  if (props.headerRender === false || props.pure) {
    return null;
  }
  return <Header {...props} />;
};

const footerRender = (props: BasicLayoutProps): React.ReactNode => {
  if (props.footerRender === false || props.pure) {
    return null;
  }
  if (props.footerRender) {
    return props.footerRender({ ...props }, <Footer />);
  }
  return null;
};

const renderSiderMenu = (props: BasicLayoutProps): React.ReactNode => {
  const { layout, isMobile, menuRender } = props;
  if (props.menuRender === false || props.pure) {
    return null;
  }
  if (layout === 'top' && !isMobile) {
    return <SiderMenu {...props} hide />;
  }
  if (menuRender) {
    return menuRender(props, <SiderMenu {...props} />);
  }

  return <SiderMenu {...props} />;
};

const defaultPageTitleRender = (
  pageProps: GetPageTitleProps,
  props: BasicLayoutProps,
): {
  title: string;
  id: string;
  pageName: string;
} => {
  const { pageTitleRender } = props;
  const pageTitleInfo = getPageTitleInfo(pageProps);
  if (pageTitleRender === false) {
    return {
      title: props.title || '',
      id: '',
      pageName: '',
    };
  }
  if (pageTitleRender) {
    const title = pageTitleRender(
      pageProps,
      pageTitleInfo.title,
      pageTitleInfo,
    );
    if (typeof title === 'string') {
      return {
        ...pageTitleInfo,
        title,
      };
    }
    warning(
      typeof title === 'string',
      'nbe-layout: renderPageTitle return value should be a string',
    );
  }
  return pageTitleInfo;
};

export type BasicLayoutContext = { [K in 'location']: BasicLayoutProps[K] } & {
  breadcrumb: { [path: string]: MenuDataItem };
};

const getPaddingLeft = (
  hasLeftPadding: boolean,
  collapsed: boolean | undefined,
  siderWidth: number,
): number | undefined => {
  if (hasLeftPadding) {
    return collapsed ? 48 : siderWidth;
  }
  return 0;
};

/**
 * üåÉ Powerful and easy to use beautiful layout
 * üèÑ‚Äç Support multiple topics and layout types
 * @param props
 */
const BasicLayout: React.FC<BasicLayoutProps> = (props) => {
  const {
    children,
    onCollapse: propsOnCollapse,
    location = { pathname: '/' },
    fixSiderbar,
    navTheme,
    contentStyle,
    route = {
      routes: [],
    },
    layout: defaultPropsLayout,
    style,
    disableContentMargin,
    siderWidth = 208,
    menu,
    isChildrenLayout: propsIsChildrenLayout,
    menuDataRender,
    loading,
    ...rest
  } = props;
  const propsLayout = compatibleLayout(defaultPropsLayout);
  const { prefixCls } = rest;
  const formatMessage = ({
    id,
    defaultMessage,
    ...restParams
  }: {
    id: string;
    defaultMessage?: string;
  }): string => {
    if (props.formatMessage) {
      return props.formatMessage({
        id,
        defaultMessage,
        ...restParams,
      });
    }
    const locales = getLocales();
    if (locales[id]) {
      return locales[id];
    }
    if (defaultMessage) {
      return defaultMessage as string;
    }
    return id;
  };

  const colSize = useAntdMediaQuery();
  const { routes = [] } = route;
  const [menuInfoData, setMenuInfoData] = useMergeValue<{
    breadcrumb?: {
      [key: string]: MenuDataItem;
    };
    breadcrumbMap?: Map<string, MenuDataItem>;
    menuData?: MenuDataItem[];
  }>(() => getMenuData(routes, menu, formatMessage, menuDataRender));

  let renderMenuInfoData: {
    breadcrumb?: {
      [key: string]: MenuDataItem;
    };
    breadcrumbMap?: Map<string, MenuDataItem>;
    menuData?: MenuDataItem[];
  } = {};

  // Si existe menuDataRender, debe mostrarse cada vez, de lo contrario no se puede garantizar la sincronizaci√≥n de datos
  if (menuDataRender) {
    renderMenuInfoData = getMenuData(
      routes,
      menu,
      formatMessage,
      menuDataRender,
    );
  }

  const isMobile =
    (colSize === 'sm' || colSize === 'xs') && !props.disableMobile;

  const { breadcrumb = {}, breadcrumbMap, menuData = [] } = !menuDataRender
    ? menuInfoData
    : renderMenuInfoData;
  /**
   *  Si menuRender no existe, puede hacer una optimizaci√≥n de rendimiento
   *  Mientras los enrutadores no est√©n actualizados, no es necesario volver a calcular
   */
  useDeepCompareEffect(() => {
    if (!menuDataRender) {
      const infoData = getMenuData(routes, menu, formatMessage, menuDataRender);

      const animationFrameId = requestAnimationFrame(() => {
        setMenuInfoData(infoData);
      });
      return () =>
        window.cancelAnimationFrame &&
        window.cancelAnimationFrame(animationFrameId);
    }
    return () => null;
  }, [props.route, stringify(menu)]);

  // If it is a fix menu, calculate padding
  // don't need padding in phone mode
  const hasLeftPadding = propsLayout !== 'top' && !isMobile;

  const [collapsed, onCollapse] = useMergeValue<boolean>(false, {
    value: props.collapsed,
    onChange: propsOnCollapse,
  });

  // Splicing parameters, adding menuData and formatMessage in props
  const defaultProps = Omit(
    {
      ...props,
      formatMessage,
      breadcrumb,
      layout: compatibleLayout(props.layout) as 'side',
    },
    ['className', 'style'],
  );

  // gen page title
  const pageTitleInfo = defaultPageTitleRender(
    {
      pathname: location.pathname,
      ...defaultProps,
      breadcrumbMap,
    },
    props,
  );

  // gen breadcrumbProps, parameter for pageHeader
  const breadcrumbProps = getBreadcrumbProps({
    ...defaultProps,
    breadcrumbMap,
  });

  // render sider dom
  const siderMenuDom = renderSiderMenu({
    ...defaultProps,
    menuData,
    onCollapse,
    isMobile,
    theme: (navTheme || 'light').toLocaleLowerCase().includes('light')
      ? 'dark'
      : 'light',
    collapsed,
  });

  // render header dom
  const headerDom = headerRender({
    ...defaultProps,
    hasSiderMenu: !!siderMenuDom,
    menuData,
    isMobile,
    collapsed,
    onCollapse,
    theme: (navTheme || 'dark').toLocaleLowerCase().includes('dark')
      ? 'dark'
      : 'light',
  });

  // render footer dom
  const footerDom = footerRender({
    isMobile,
    collapsed,
    ...defaultProps,
  });

  const { isChildrenLayout: contextIsChildrenLayout } = useContext(
    RouteContext,
  );

  // Si se define en par√°metros, los par√°metros prevalecer√°n
  const isChildrenLayout =
    propsIsChildrenLayout !== undefined
      ? propsIsChildrenLayout
      : contextIsChildrenLayout;

  const baseClassName = `${prefixCls}-basicLayout`;
  // gen className
  const className = classNames(
    props.className,
    'ant-design-pro',
    baseClassName,
    {
      [`screen-${colSize}`]: colSize,
      [`${baseClassName}-top-menu`]: propsLayout === 'top',
      [`${baseClassName}-is-children`]: isChildrenLayout,
      [`${baseClassName}-fix-siderbar`]: fixSiderbar,
      [`${baseClassName}-mobile`]: isMobile,
    },
  );

  /**
   * Calcule el ancho del control deslizante
   */
  const leftSiderWidth = getPaddingLeft(
    !!hasLeftPadding,
    collapsed,
    siderWidth,
  );

  // Cuando siderMenuDom est√° vac√≠o, no se necesita padding
  const genLayoutStyle: CSSProperties = {
    position: 'relative',
  };

  // if is some layout childrenÔºådon't need min height
  if (isChildrenLayout || (contentStyle && contentStyle.minHeight)) {
    genLayoutStyle.minHeight = 0;
  }

  const contentClassName = classNames(`${baseClassName}-content`, {
    [`${baseClassName}-has-header`]: headerDom,
    [`${baseClassName}-content-disable-margin`]: disableContentMargin,
  });

  /**
   * Se activa cuando se cambia la p√°gina
   */
  useEffect(() => {
    const { onPageChange } = props;
    if (onPageChange) {
      onPageChange(props.location);
    }
  }, [stringify(props.location)]);
  const [hasFooterToolbar, setHasFooterToolbar] = useState(false);

  useDocumentTitle(pageTitleInfo, props.title);

  return (
    <MenuCounter.Provider>
      <RouteContext.Provider
        value={{
          ...defaultProps,
          breadcrumb: breadcrumbProps,
          menuData,
          isMobile,
          collapsed,
          isChildrenLayout: true,
          title: pageTitleInfo.pageName,
          hasSiderMenu: !!siderMenuDom,
          hasHeader: !!headerDom,
          siderWidth: leftSiderWidth,
          hasFooter: !!footerDom,
          hasFooterToolbar,
          setHasFooterToolbar,
          pageTitleInfo,
        }}
      >
        <div className={className}>
          <Layout
            style={{
              minHeight: '100%',
              ...style,
            }}
            hasSider
          >
            {siderMenuDom}
            <Layout style={genLayoutStyle}>
              {headerDom}
              <WrapContent
                isChildrenLayout={isChildrenLayout}
                {...rest}
                className={contentClassName}
                style={contentStyle}
              >
                {loading ? <PageLoading /> : children}
              </WrapContent>
              {footerDom}
            </Layout>
          </Layout>
        </div>
      </RouteContext.Provider>
    </MenuCounter.Provider>
  );
};

BasicLayout.defaultProps = {
  logo:
    'https://raw.githubusercontent.com/machinaai/pro-layout/nbe-layout/src/assets/logo/nbe-logo.svg',
  ...defaultSettings,
  prefixCls: 'ant-pro',
  siderWidth: 220,
  location: isBrowser() ? window.location : undefined,
};
export default BasicLayout;
