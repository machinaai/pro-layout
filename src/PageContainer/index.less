@import '~antd/es/style/themes/default.less';

@nbe-layout-page-container: ~'@{ant-prefix}-pro-page-container';

.@{nbe-layout-page-container}-children-content {
  margin: 24px 24px 0;
}

.@{nbe-layout-page-container} {
  &-warp {
    background-color: #f4f6fe;
    .@{ant-prefix}-tabs-nav {
      margin: 0;
    }
  }
  &-ghost {
    .@{nbe-layout-page-container}-warp {
      background-color: transparent;
    }
  }
}

.@{nbe-layout-page-container}-main {
  .@{nbe-layout-page-container}-detail {
    display: flex;
  }

  .@{nbe-layout-page-container}-row {
    display: flex;
    width: 100%;
  }

  .@{nbe-layout-page-container}-title-content {
    margin-bottom: 16px;
  }

  .@{nbe-layout-page-container}-title,
  .@{nbe-layout-page-container}-content {
    flex: auto;
  }

  .@{nbe-layout-page-container}-extraContent,
  .@{nbe-layout-page-container}-main {
    flex: 0 1 auto;
  }

  .@{nbe-layout-page-container}-main {
    width: 100%;
  }

  .@{nbe-layout-page-container}-title {
    margin-bottom: 16px;
  }

  .@{nbe-layout-page-container}-logo {
    margin-bottom: 16px;
  }

  .@{nbe-layout-page-container}-extraContent {
    min-width: 242px;
    margin-left: 88px;
    text-align: right;
  }
}

@media screen and (max-width: @screen-xl) {
  .@{nbe-layout-page-container}-main {
    .@{nbe-layout-page-container}-extraContent {
      margin-left: 44px;
    }
  }
}

@media screen and (max-width: @screen-lg) {
  .@{nbe-layout-page-container}-main {
    .@{nbe-layout-page-container}-extraContent {
      margin-left: 20px;
    }
  }
}

@media screen and (max-width: @screen-md) {
  .@{nbe-layout-page-container}-main {
    .@{nbe-layout-page-container}-row {
      display: block;
    }

    .@{nbe-layout-page-container}-action,
    .@{nbe-layout-page-container}-extraContent {
      margin-left: 0;
      text-align: left;
    }
  }
}

@media screen and (max-width: @screen-sm) {
  .@{nbe-layout-page-container}-detail {
    display: block;
  }
  .@{nbe-layout-page-container}-extraContent {
    margin-left: 0;
  }
}
