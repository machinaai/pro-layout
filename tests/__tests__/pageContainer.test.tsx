import { render, mount } from 'enzyme';
import React from 'react';
import PageContainer from '../../src/PageContainer';
import FooterToolbar from '../../src/FooterToolbar';
import BasicLayout, { BasicLayoutProps } from '../../src/BasicLayout';
import { waitForComponentToPaint } from './util';

describe('PageContainer', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: jest.fn(() => ({
        matches: false,
        addListener() {},
        removeListener() {},
      })),
    });
  });

  it('💄 base use', async () => {
    const html = render(<PageContainer title="Daniel Gomez" />);
    expect(html).toMatchSnapshot();
  });

  it('⚡️ support footer', async () => {
    const html = render(
      <PageContainer
        title="Daniel Gomez"
        footer={[<button type="button">right</button>]}
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('🔥 support footer', async () => {
    const html = render(
      <PageContainer
        title="Daniel Gomez"
        breadcrumb={{
          routes: [
            {
              path: '/',
              breadcrumbName: 'home',
            },
          ],
        }}
        footer={[
          <button key="right" type="button">
            right
          </button>,
        ]}
      />,
    );
    expect(html).toMatchSnapshot();
  });

  it('🔥 footer bar support extra', async () => {
    const html = render(
      <FooterToolbar
        className="qixian_footer"
        extra={
          <img
            src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
            alt="logo"
          />
        }
      >
        <button type="button">right</button>
      </FooterToolbar>,
    );
    expect(html).toMatchSnapshot();
  });

  it('🔥 footer bar support renderContent', async () => {
    const html = render(
      <FooterToolbar
        className="qixian_footer"
        extra={
          <img
            src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
            alt="logo"
          />
        }
        renderContent={() => {
          return 'home_toolbar';
        }}
      >
        <button type="button">right</button>
      </FooterToolbar>,
    );
    expect(html).toMatchSnapshot();
  });

  it('🐲 footer should know width', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout>
        <PageContainer
          footer={[
            <button type="button" key="button">
              qixian
            </button>,
          ]}
        />
      </BasicLayout>,
    );
    await waitForComponentToPaint(wrapper);

    expect(wrapper?.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe(
      'calc(100% - 220px)',
    );
    wrapper.setProps({
      collapsed: true,
    });

    await waitForComponentToPaint(wrapper);

    expect(wrapper?.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe(
      'calc(100% - 48px)',
    );

    wrapper.setProps({
      layout: 'top',
    });

    expect(wrapper?.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe(
      '100%',
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('🐲 FooterToolbar should know width', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout>
        <PageContainer>
          <FooterToolbar>
            <button type="button" key="button">
              qixian
            </button>
          </FooterToolbar>
        </PageContainer>
      </BasicLayout>,
    );
    await waitForComponentToPaint(wrapper);

    expect(wrapper?.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe(
      'calc(100% - 220px)',
    );
    wrapper.setProps({
      collapsed: true,
    });

    await waitForComponentToPaint(wrapper);

    expect(wrapper.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe(
      'calc(100% - 48px)',
    );

    wrapper.setProps({
      layout: 'top',
    });

    expect(wrapper.find('.ant-pro-footer-bar')?.props()?.style?.width).toBe(
      '100%',
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('🐲 footer is null, do not render footerToolbar ', async () => {
    const wrapper = mount(
      <PageContainer
        footer={[
          <button type="button" key="button">
            qixian
          </button>,
        ]}
      />,
    );
    await waitForComponentToPaint(wrapper);
    expect(wrapper.render()).toMatchSnapshot();

    wrapper.setProps({
      footer: undefined,
    });
    await waitForComponentToPaint(wrapper);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
