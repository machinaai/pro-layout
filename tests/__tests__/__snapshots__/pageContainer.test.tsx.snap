// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer ‚ö°Ô∏è support footer 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="Daniel Gomez"
          >
            Daniel Gomez
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
  <div
    class="ant-pro-footer-bar"
  >
    <div
      class="ant-pro-footer-bar-left"
    />
    <div
      class="ant-pro-footer-bar-right"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <button
            type="button"
          >
            right
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageContainer üê≤ FooterToolbar should know width 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout screen-md ant-pro-basicLayout-top-menu"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="min-height: 100%;"
  >
    <section
      class="ant-layout"
      style="position: relative;"
    >
      <header
        class="ant-layout-header ant-pro-top-menu"
        style="padding: 0px; height: 64px; line-height: 64px; width: 100%; z-index: 9;"
      >
        <div
          class="ant-pro-top-nav-header"
        >
          <div
            class="ant-pro-top-nav-header-main "
          >
            <div
              class="ant-pro-top-nav-header-main-left"
            >
              <div
                class="ant-pro-top-nav-header-logo"
                id="logo"
              >
                <a>
                  <img
                    alt="logo"
                    src="https://raw.githubusercontent.com/machinaai/pro-layout/nbe-layout/src/assets/logo/nbe-logo.svg"
                  />
                  <h1>
                    React Designer
                  </h1>
                </a>
              </div>
            </div>
            <div
              class="ant-pro-top-nav-header-menu"
              style="flex: 1;"
            >
              <ul
                class="ant-menu top-nav-menu ant-menu-dark ant-menu-root ant-menu-horizontal"
                role="menu"
              />
            </div>
          </div>
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-page-container"
        >
          <div
            class="ant-pro-page-container-warp"
          >
            <div
              class="ant-page-header ant-page-header-ghost"
            >
              <div
                class="ant-page-header-heading"
              >
                <div
                  class="ant-page-header-heading-left"
                >
                  <span
                    class="ant-page-header-heading-title"
                    title="React Designer"
                  >
                    React Designer
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-pro-grid-content"
          >
            <div
              class="ant-pro-grid-content-children"
            >
              <div>
                <div
                  class="ant-pro-page-container-children-content"
                >
                  <div
                    class="ant-pro-footer-bar"
                    style="width: 100%;"
                  >
                    <div
                      class="ant-pro-footer-bar-left"
                    />
                    <div
                      class="ant-pro-footer-bar-right"
                    >
                      <div
                        class="ant-space ant-space-horizontal ant-space-align-center"
                      >
                        <div
                          class="ant-space-item"
                        >
                          <button
                            type="button"
                          >
                            qixian
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  style="height: 48px; margin-top: 24px;"
                />
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
  </section>
</div>
`;

exports[`PageContainer üê≤ footer is null, do not render footerToolbar  1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    />
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
  <div
    class="ant-pro-footer-bar"
  >
    <div
      class="ant-pro-footer-bar-left"
    />
    <div
      class="ant-pro-footer-bar-right"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <button
            type="button"
          >
            qixian
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageContainer üê≤ footer is null, do not render footerToolbar  2`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    />
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer üê≤ footer should know width 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout screen-md ant-pro-basicLayout-top-menu"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="min-height: 100%;"
  >
    <section
      class="ant-layout"
      style="position: relative;"
    >
      <header
        class="ant-layout-header ant-pro-top-menu"
        style="padding: 0px; height: 64px; line-height: 64px; width: 100%; z-index: 9;"
      >
        <div
          class="ant-pro-top-nav-header"
        >
          <div
            class="ant-pro-top-nav-header-main "
          >
            <div
              class="ant-pro-top-nav-header-main-left"
            >
              <div
                class="ant-pro-top-nav-header-logo"
                id="logo"
              >
                <a>
                  <img
                    alt="logo"
                    src="https://raw.githubusercontent.com/machinaai/pro-layout/nbe-layout/src/assets/logo/nbe-logo.svg"
                  />
                  <h1>
                    React Designer
                  </h1>
                </a>
              </div>
            </div>
            <div
              class="ant-pro-top-nav-header-menu"
              style="flex: 1;"
            >
              <ul
                class="ant-menu top-nav-menu ant-menu-dark ant-menu-root ant-menu-horizontal"
                role="menu"
              />
            </div>
          </div>
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-page-container"
        >
          <div
            class="ant-pro-page-container-warp"
          >
            <div
              class="ant-page-header ant-page-header-ghost"
            >
              <div
                class="ant-page-header-heading"
              >
                <div
                  class="ant-page-header-heading-left"
                >
                  <span
                    class="ant-page-header-heading-title"
                    title="React Designer"
                  >
                    React Designer
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-pro-grid-content"
          >
            <div
              class="ant-pro-grid-content-children"
            />
          </div>
          <div
            class="ant-pro-footer-bar"
            style="width: 100%;"
          >
            <div
              class="ant-pro-footer-bar-left"
            />
            <div
              class="ant-pro-footer-bar-right"
            >
              <div
                class="ant-space ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                >
                  <button
                    type="button"
                  >
                    qixian
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
  </section>
</div>
`;

exports[`PageContainer üíÑ base use 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="Daniel Gomez"
          >
            Daniel Gomez
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
</div>
`;

exports[`PageContainer üî• footer bar support extra 1`] = `
<div
  class="qixian_footer ant-pro-footer-bar"
>
  <div
    class="ant-pro-footer-bar-left"
  >
    <img
      alt="logo"
      src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
    />
  </div>
  <div
    class="ant-pro-footer-bar-right"
  >
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
      >
        <button
          type="button"
        >
          right
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageContainer üî• footer bar support renderContent 1`] = `
<div
  class="qixian_footer ant-pro-footer-bar"
>
  home_toolbar
</div>
`;

exports[`PageContainer üî• support footer 1`] = `
<div
  class="ant-pro-page-container"
>
  <div
    class="ant-pro-page-container-warp"
  >
    <div
      class="ant-page-header has-breadcrumb ant-page-header-ghost"
    >
      <div
        class="ant-breadcrumb"
      >
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            <span>
              home
            </span>
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
      </div>
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="Daniel Gomez"
          >
            Daniel Gomez
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-pro-grid-content"
  >
    <div
      class="ant-pro-grid-content-children"
    />
  </div>
  <div
    class="ant-pro-footer-bar"
  >
    <div
      class="ant-pro-footer-bar-left"
    />
    <div
      class="ant-pro-footer-bar-right"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <button
            type="button"
          >
            right
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
